FirebaseStorage/Sources/FIRStorageUploadTask.m

```diff
@@ -193,11 +193,14 @@ - (BOOL)isContentToUploadValid:(NSError **)outError {
   }
 
   NSError *fileReachabilityError;
-  if (![_fileURL checkResourceIsReachableAndReturnError:&fileReachabilityError]) {
+  if (![_fileURL checkResourceIsReachableAndReturnError:&fileReachabilityError] ||
+      ![self fileURLisFile:_fileURL]) {
```

Comment:
The [`checkResourceIsReachableAndReturnError`](https://developer.apple.com/documentation/foundation/nsurl/1410411-checkpromiseditemisreachableandr?language=objc) returns true or false depending on if the `_fileURL` can be reached. This is why attempts to upload directories weren't being caught in the first place, because they are reachable. Instead of doing a check in `putFile:`, I figured I should move the `fileURLisFile` helper method to  `FIRStorageUploadTask.m` and call it in this `isContentToUploadValid` method so we have a single check to verify that the file we are trying to upload is both a reachable resource and **it is a file**.